pkg/signing/signing_test.go

```diff
@@ -0,0 +1,210 @@
+				}
+			},
+			wantErr: "",
+			expectedSignature: "6ddb61a19b9df71136b48c80b2e86e7e20313d5eec0de9210802335b3" +
+				"00ba8df6c332d35a5d753a028d703769fd9b66d7ce5902d80369750cf55118b1679d84900",
+		},
+		{
+			name: "should fail with empty file",
+			setup: func() (filename string, cleanup func()) {
+				tmpFile, _ := os.CreateTemp("", "test_file")
+				name := tmpFile.Name()
+				err := tmpFile.Close()
+				require.NoError(t, err, "Error closing file")
+				return name, func() {
+					err = os.Remove(name)
+					require.NoError(t, err, "Error removing file")
+				}
+			},
+			wantErr: "error with file: content is empty",
+		},
+		{
+			name: "should fail with non-existent file",
+			setup: func() (filename string, cleanup func()) {
+				tmpFile, _ := os.CreateTemp("", "test_file")
+				name := tmpFile.Name()
+				err := tmpFile.Close()
+				require.NoError(t, err, "Error closing file")
+				err = os.Remove(name)
+				require.NoError(t, err, "Error removing file")
+				return name, func() {}
+			},
+			wantErr: "error reading [file=",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			filename, cleanup := tc.setup()
+			defer cleanup()
+
+			signatureBytes, err := signer.SignFile(filename)
+			signature := SignatureBytesToHex(signatureBytes)
+			if tc.wantErr != "" {
+				require.Error(t, err, "Expected an error for %v", tc.name)
+				require.Contains(t, err.Error(), tc.wantErr, "SignFile() error = %v, wantErr %v", err, tc.wantErr)
+			} else {
+				require.NoError(t, err, "SignFile() unexpected error = %v", err)
+				require.Equal(t, tc.expectedSignature, signature, "Signature mismatch")
+			}
+		})
+	}
+}
+
+func TestSignBytes(t *testing.T) {
+	privateKey, _ := HexToECDSA("59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d")
+	signer := NewSigner(privateKey)
+
+	testCases := []struct {
+		name              string
+		content           []byte
+		wantErr           string
+		expectedSignature string
+	}{
+		{
+			name:    "should sign bytes",
+			content: []byte("data to be signed"),
+			wantErr: "",
+			expectedSignature: "6ddb61a19b9df71136b48c80b2e86e7e20313d5eec0de9210802335b3" +
+				"00ba8df6c332d35a5d753a028d703769fd9b66d7ce5902d80369750cf55118b1679d84900",
+		},
+		{
+			name:    "should fail with empty bytes",
+			content: []byte(""),
+			wantErr: "error with data: content is empty",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			signatureBytes, err := signer.SignBytes(tc.content)
+			signature := SignatureBytesToHex(signatureBytes)
+			if tc.wantErr != "" {
+				require.Error(t, err, "Expected an error for %v", tc.name)
+				require.Contains(t, err.Error(), tc.wantErr, "SignBytes() error = %v, wantErr %v", err, tc.wantErr)
+			} else {
+				require.NoError(t, err, "SignBytes() unexpected error = %v", err)
+				require.Equal(t, tc.expectedSignature, signature, "Signature mismatch")
+			}
+		})
+	}
+}
+
+func TestPrivateKey(t *testing.T) {
```

Comment:
Not sure but this test. `HexToECDSA` and `FileToECDSA` are just calling std lib methods that are well-tested. This test does not add more signal. Maybe just the happy path test is good enough because it tests that the method calls the correct std library method. But all the other error tests are kind of redundant, i guess